rebuild/src/signature.h

```diff
@@ -0,0 +1,119 @@
+#ifndef BLST_TS_SIGNATURE_H__
+#define BLST_TS_SIGNATURE_H__
+
+#include <memory>
+
+#include "addon.h"
+#include "blst.hpp"
+#include "napi.h"
+
+#define BLST_TS_SIGNATURE_LENGTH_COMPRESSED 96U
+#define BLST_TS_SIGNATURE_LENGTH_UNCOMPRESSED 192U
+
+typedef struct {
+    std::unique_ptr<blst::P2_Affine> smart_pointer;
+    blst::P2_Affine *raw_point;
+} P2AffineGroup;
+
+class P2Wrapper {
+   public:
+    virtual ~P2Wrapper() = default;
+    virtual bool IsInfinite() = 0;
+    virtual bool InGroup() = 0;
+    virtual void Serialize(bool compress, blst::byte *out) = 0;
+    virtual void AddTo(blst::P2 &point) = 0;
+    virtual blst::P2 MultiplyBy(
+        blst::byte *rand_bytes, size_t rand_bytes_length) = 0;
+    virtual P2AffineGroup AsAffine() = 0;
+    virtual void Sign(
+        const blst::SecretKey &key,
+        uint8_t *msg,
+        size_t msg_length,
+        const std::string &dst) = 0;
+};
+
+class P2 : public P2Wrapper {
+    blst::P2 _point;
+
+   public:
+    P2(blst::P2 point) : _point(std::move(point)) {}
+    bool IsInfinite() override { return _point.is_inf(); }
```

Comment:
`final` would probably a better keyword here, I assume you don't want to inherit from `P2`.

Reply:
Updated. 3d138e599a29adef3b25520fb6f008257f08657d